__tests__/Unit/pages/Dashboard/dashboardTest.test.tsx

```diff
@@ -0,0 +1,56 @@
+import DashboardPage from '@/pages/dashboard';
+import { renderWithRouter } from '@/test_utils/createMockRouter';
+import { Provider } from 'react-redux';
+import { store } from '@/app/store';
+import { fireEvent, screen } from '@testing-library/react';
+
+describe('dashboard page test', function () {
+    it('checks if the page is rendered with exact components', function () {
+        renderWithRouter(
+            <Provider store={store()}>
+                <DashboardPage />
+            </Provider>,
+            { query: { dev: 'true' } }
+        );
+
+        const searchBar = screen.getByTestId('searchbar_input');
+        expect(searchBar).toBeInTheDocument();
+
+        const searchButton = screen.getByTestId('search_btn');
+        expect(searchButton.innerHTML).toBe('Search');
+    });
+
+    it('renders 404 without passing the feature flag', function () {
+        renderWithRouter(
+            <Provider store={store()}>
+                <DashboardPage />
+            </Provider>
+        );
+
+        const headings = screen.getAllByRole('heading');
+        expect(headings).toHaveLength(1);
+        expect(headings[0].innerHTML).toBe('404 - Page Not Found');
+    });
+
+    it('console logs the value', function () {
+        console.log = jest.fn();
```

Comment:
Are we testing the console.log here and why?

Reply:
yes because the current version of splitNSearch console logs the plotted value, this helps me check if the value I enter is split based on comma